Formula/macchina.rb

```diff
@@ -0,0 +1,17 @@
+class Macchina < Formula
+  desc "Basic system information fetcher, with a focus on performance and minimalism"
+  homepage "https://github.com/Macchina-CLI/macchina"
+  url "https://github.com/Macchina-CLI/macchina/archive/v0.7.0.tar.gz"
+  sha256 "6ff1497864a400f5eade2b46984cbe1259a0f7a75bf7191a2a75e111a8ffe119"
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "macchina", "--doctor"
```

Comment:
Can you test some output here?

Reply:
The '--doctor' function is very platform specific and checks for general warnings & errors. Even if there are errors, it might only mean that the person does not have a battery installed, causing a function to fail. I don't think checking for a specific message in the output makes sense, but it is a great test to see if the executable returns an exit code that marks the successful execution (even if a platform-specific function is not available). If the application crashes (e.g. due to a segmentation fault, **not** because we couldn't query a information from the system), we will see it here because "--doctor" checks every functionality.

Reply:
<img width="1052" alt="image" src="https://user-images.githubusercontent.com/7266447/115446459-8ab50a00-a217-11eb-8cea-e82081f247a1.png">

Reply:
So we can do this then
```rb
assert_match "Let's check your system for errors...", shell_output("#{bin}/macchina --doctor")
```

Reply:
Hi!

Author here, you can do that, yes.